src/main/java/LadderGame.java

```diff
@@ -11,32 +16,62 @@ private LadderGame() {
     }
 
     public static void run() {
-        Names names = readNames();
+        Names names = repeat(() -> new Names(InputView.readPersonNames()));
         OutputView.printNewLine();
-        Height height = readHeight();
+        Prizes prizes = repeat(() -> new Prizes(InputView.readPrizes(), names));
         OutputView.printNewLine();
+        Height height = repeat(() -> new Height(InputView.readHeight()));
+        OutputView.printNewLine();
+        Ladder ladder = new Ladder(new RandomBridgeConstructStrategy(), names, prizes, height);
 
-        Ladder ladder = new Ladder(new RandomBridgeConstructStrategy(), names, height);
+        handleResult(names, prizes, ladder);
+    }
 
+    private static void handleResult(Names names, Prizes prizes, Ladder ladder) {
         OutputView.printNames(names);
         OutputView.printLadder(ladder);
+        OutputView.printPrizes(prizes);
+        OutputView.printNewLine();
+        repeat(() -> printResult(names, ladder));
+    }
+
+    private static void printResult(Names names, Ladder ladder) {
+        String target = InputView.readResultTarget();
+        if (handleAll(ladder, target)) {
```

Comment:
이거 handleAll이 조건절로 들어가기 보다는 target.eqauls("all")이 조건절로 들어가서, if문 내부에서 handleAll 과정을 수행하는 것은 어떤가요?
```suggestion
        if (target.equals("all")) {
        // all일 때 처리
        }
```